src/algorithm/mod.rs

```diff
@@ -138,6 +138,8 @@ pub(crate) fn validate_size_impl(
     let size = elements * elem_size as f64;
     let max_mega = if cfg!(target_arch = "wasm32") {
         256
+    } else if cfg!(target_arch = "x86") {
```

Comment:
This would apply to *any* `x86` architecture, which is not correct.

Reply:
I think the better option is likely to change the calculation below (`max_mega * 1024usize.pow(2)`) to use `f64`s so it doesn't overflow.

Reply:
I think it should be:
```rs
let max_mega = if cfg!(target_pointer_width = "32") {
    256
} else {
    4096
};
```

Reply:
@kaikalii I don't know rust, but thought I would see what I could do. You should absolutely go with your suggested change.

Reply:
> I think it should be:
> 
> ```rust
> let max_mega = if cfg!(target_pointer_width = "32") {
>     256
> } else {
>     4096
> };
> ```

Ya you're probably right.